yt/frontends/parthenon/data_structures.py

```diff
@@ -0,0 +1,326 @@
+    ):
+        self.fluid_types += ("parthenon",)
+        if parameters is None:
+            parameters = {}
+        self.specified_parameters = parameters
+        if units_override is None:
+            units_override = {}
+        self._handle = HDF5FileHandler(filename)
+        xrat = self._handle["Info"].attrs["RootGridDomain"][2]
+        yrat = self._handle["Info"].attrs["RootGridDomain"][5]
+        zrat = self._handle["Info"].attrs["RootGridDomain"][8]
+        if xrat != 1.0 or yrat != 1.0 or zrat != 1.0:
+            self.logarithmic = True
+            raise ValueError(
+                "Logarithmic grids not yet supported in Parthenon frontend."
+            )
+        else:
+            self._index_class = ParthenonHierarchy
+            self.logarithmic = False
+        self._magnetic_factor = get_magnetic_normalization(magnetic_normalization)
+
+        self.geometry = geom_map[self._handle["Info"].attrs["Coordinates"]]
+
+        if self.geometry == "cylindrical":
+            axis_order = ("r", "theta", "z")
+        else:
+            axis_order = None
+
+        Dataset.__init__(
+            self,
+            filename,
+            dataset_type,
+            units_override=units_override,
+            unit_system=unit_system,
+            default_species_fields=default_species_fields,
+            axis_order=axis_order,
+        )
+        if storage_filename is None:
+            storage_filename = self.basename + ".yt"
+        self.storage_filename = storage_filename
+
+    def _set_code_unit_attributes(self):
+        """
+        Generates the conversion to various physical _units based on the
+        parameter file
+        """
+        for unit, cgs in [
+            ("length", "cm"),
+            ("time", "s"),
+            ("mass", "g"),
+        ]:
+            unit_param = f"Hydro/code_{unit}_cgs"
+            # We set these to cgs for now, but they may have been overridden
+            if getattr(self, unit + "_unit", None) is not None:
+                continue
+            elif unit_param in self.parameters:
+                setattr(
+                    self, f"{unit}_unit", self.quan(self.parameters[unit_param], cgs)
+                )
+            else:
+                mylog.warning(f"Assuming 1.0 = 1.0 {cgs}")
+                setattr(self, f"{unit}_unit", self.quan(1.0, cgs))
+
+        self.magnetic_unit = np.sqrt(
+            self._magnetic_factor
+            * self.mass_unit
+            / (self.time_unit**2 * self.length_unit)
+        )
+        self.magnetic_unit.convert_to_units("gauss")
+        self.velocity_unit = self.length_unit / self.time_unit
+
+    def _parse_parameter_file(self):
+        self.parameters.update(self.specified_parameters)
+        for key, val in self._handle["Params"].attrs.items():
+            if key in self.parameters.keys():
+                mylog.warning(
+                    f"Overriding existing 'f{key}' key in ds.parameters from data 'Params'"
```

Comment:
I think the `f` is a typo ? 
```suggestion
                    f"Overriding existing key {key!r} in ds.parameters from data 'Params'"
```